engine/src/io/data_parser/sql/MySQLParser.cpp

```diff
@@ -0,0 +1,598 @@
+          }
+        } break;
+        case cudf::type_id::UINT16: {
+          uint32_t raw_data = res->getUInt(mysql_col);
+          if (res->wasNull()) {
+            valid = 0;
+          } else {
+            std::vector<uint16_t> *v = (std::vector<uint16_t>*)host_cols[col];
+            uint16_t real_data = static_cast<uint16_t>(raw_data);
+            (*v)[row] = real_data;
+          }
+        } break;
+        case cudf::type_id::UINT32: {
+          uint32_t raw_data = res->getUInt(mysql_col);
+          if (res->wasNull()) {
+            valid = 0;
+          } else {
+            std::vector<uint32_t> *v = (std::vector<uint32_t>*)host_cols[col];
+            (*v)[row] = raw_data;
+          }
+        } break;
+        case cudf::type_id::UINT64: {
+          uint64_t raw_data = res->getUInt64(mysql_col);
+          if (res->wasNull()) {
+            valid = 0;
+          } else {
+            std::vector<uint64_t> *v = (std::vector<uint64_t>*)host_cols[col];
+            (*v)[row] = raw_data;
+          }
+        } break;
+        case cudf::type_id::FLOAT32: {
+          long double raw_data = res->getDouble(mysql_col);
+          if (res->wasNull()) {
+            valid = 0;
+          } else {
+            std::vector<float> *v = (std::vector<float>*)host_cols[col];
+            float real_data = static_cast<float>(raw_data);
+            (*v)[row] = real_data;
+          }
+        } break;
+        case cudf::type_id::FLOAT64: {
+          long double raw_data = res->getDouble(mysql_col);
+          if (res->wasNull()) {
+            valid = 0;
+          } else {
+            std::vector<double> *v = (std::vector<double>*)host_cols[col];
+            double real_data = static_cast<double>(raw_data);
+            (*v)[row] = real_data;
+          }
+        } break;
+        case cudf::type_id::BOOL8: {
+          bool raw_data = res->getBoolean(mysql_col);
+          if (res->wasNull()) {
+            valid = 0;
+          } else {
+            std::vector<int8_t> *v = (std::vector<int8_t>*)host_cols[col];
+            int8_t real_data = static_cast<int8_t>(raw_data);
+            (*v)[row] = real_data;
+          }
+        } break;
+        case cudf::type_id::TIMESTAMP_DAYS:
+        case cudf::type_id::TIMESTAMP_SECONDS:
+        case cudf::type_id::TIMESTAMP_MILLISECONDS:
+        case cudf::type_id::TIMESTAMP_MICROSECONDS:
+        case cudf::type_id::TIMESTAMP_NANOSECONDS:
+        // TODO percy it seems we don't support this yet
+        // case cudf::type_id::DURATION_DAYS: {} break;
+        // case cudf::type_id::DURATION_SECONDS: {} break;
+        // case cudf::type_id::DURATION_MILLISECONDS: {} break;
+        // case cudf::type_id::DURATION_MICROSECONDS: {} break;
+        // case cudf::type_id::DURATION_NANOSECONDS: {} break;
+        // case cudf::type_id::DICTIONARY32: {} break;
+        case cudf::type_id::STRING: {
+          std::string real_data = res->getString(mysql_col).asStdString();
```

Comment:
it might be faster to accumulate a vector of strings and make a cudf string column once you have a batch full

Reply:
I think that is what we are doing, we create and accumulate in host and the once is ready we create the cudf columns, perhaps we need more info about what do you have in mind here cc @gcca